jetbrains-core/src/software/aws/toolkits/jetbrains/core/region/AwsRegionProvider.kt

```diff
@@ -14,10 +15,16 @@ import software.aws.toolkits.core.region.ServiceEndpointResource
 import software.aws.toolkits.core.region.ToolkitRegionProvider
 import software.aws.toolkits.core.utils.getLogger
 import software.aws.toolkits.core.utils.inputStream
+import software.aws.toolkits.core.utils.onNull
+import software.aws.toolkits.core.utils.tryOrNull
 import software.aws.toolkits.core.utils.warn
 import software.aws.toolkits.jetbrains.core.RemoteResourceResolverProvider
 
 class AwsRegionProvider constructor(remoteResourceResolverProvider: RemoteResourceResolverProvider) : ToolkitRegionProvider() {
+    private val regionChain by lazy {
+        // Querying the instance metadata is expensive due to high timeouts and retries
```

Comment:
They fixed this in the SDK I believe, we should recheck this one day...

Reply:
Yeah and if it isn't fixed - do it manually - I want us to create an EC2 credential provider too - cos I think EC2 should "just work"